File: nattybear/level1/배열-자르기/solution.js
function solution(n, left, right) {
  const arr = []
  const remainder = left % n
  const limit = (right - left + 1) + remainder
  let cnt = 0
  for (let i = row(n, left); i <= n; i++) {
    let j = 1
    for (; j <= i; j++) {
      arr.push(i)
      cnt++
      if (cnt == limit)
        return arr.slice(remainder)
    }
    for (; j <= n; j++) {
      arr.push(j)
      cnt++
      if (cnt == limit)
        return arr.slice(remainder)
    }
  }
}

function row(n, left) {
  return Math.floor(left / n) + 1
}

File: nattybear/level1/배열-자르기/Solution.java
import java.util.ArrayList;
import java.util.stream.LongStream;

class Solution {
  public int[] solution(int n, long left, long right) {
    ArrayList<Integer> list = new ArrayList<>();
    int remainder = (int) (left % n);
    long limit = (right - left + 1) + remainder;
    long cnt = 0;
    for (int i = row(n, left); i <= n; i++) {
      int j = 1;
      for (; j <= i; j++) {
        list.add(i);
        cnt++;
        if (cnt == limit)
          return list.stream()
            .mapToInt(Integer::intValue)
            .skip(remainder).toArray();
      }
      for (; j <= n; j++) {
        list.add(j);
        cnt++;
        if (cnt == limit)
          return list.stream()
            .mapToInt(Integer::intValue)
            .skip(remainder).toArray();
      }
    }
    return list.stream()
      .mapToInt(Integer::intValue)
      .toArray();
  }

  static int row(int n, long left) {
    long _row = left / (long) n + 1;
    return (int) _row;
  }
}
